### PR TITLE
Files for readthedocs

### PR DESCRIPTION
**Pull request by tfalders**
Partially addresses SWDEV-268307.

Emails about adding hipSOLVER to the ROCm documentation indicated that a readthedocs webpage should be created for hipSOLVER. I've created a page that currently points to the readthedocs branch of my local fork:
https://hipsolver.readthedocs.io/en/latest/index.html

### CODE REVIEW

**Review by cgmb — COMMENTED** [review]
The README's link to the ROCm install guide redirects to the main rocmdocs page. I think it's out-of-date. This is the [ROCm Install Guide](https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html).

Interesting use of m2d2. That should save some effort. When I was making docs for PyPI, I used `pandoc` to manually convert from markdown to rst, but doing it on-the-fly is even better.

Also, just a heads up that the docs on building the software may change as we move to Python for our helper utilities.

**Review by cgmb — APPROVED** [review]
I'll add a CI job later.

**Comment by cgmb** [comment]
I've added a working CI job to this PR. I don't have a good way to share it, but if you want a button to navigate from the PR to the docs for a PR, you can create a bookmark with this as the URL:
```
javascript:(function() {var m = window.location.toString().match(/\/([^/]+)\/pull\/(\d+)/);window.open(`http://math-ci.rocm.amd.com/job/compute-rocm-dkms-no-npi-hipclang/job/Static%20Analysis/job/${m[1]}/job/PR-${m[2]}/Documentation/`);})();
```

**Review by jzuniga-amd — COMMENTED** [review]
Looks good... few minor concerns. 
(I added some suggested changes in a PR to your branch, please have a look)

File: bump_hipsolver_version.sh [hunk 1]
```diff
@@ -8,12 +8,16 @@
 OLD_HIPSOLVER_VERSION="1.0.0"
 NEW_HIPSOLVER_VERSION="1.1.0"
 
+OLD_HIPSOLVER_DOCS_VERSION="3.13"
+NEW_HIPSOLVER_DOCS_VERSION="3.14"
```
[the remark this answers is not in the document]
Comment (tfalders) [reply bump_hipsolver_version.sh:12]: Whoops! Apparently I forgot to change the numbers when I copied it from rocSOLVER. I'll fix it.

File: README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 hipSOLVER is a LAPACK marshalling library, with multiple supported backends.  It sits between the application and a 'worker' LAPACK library, marshalling inputs into the backend library and marshalling results back to the application.  hipSOLVER exports an interface that does not require the client to change, regardless of the chosen backend.  Currently, hipSOLVER supports [rocSOLVER](https://github.com/ROCmSoftwarePlatform/rocSOLVER) and [cuSOLVER](https://developer.nvidia.com/cusolver) as backends.
 
 ## Installing Pre-built Packages
-Download pre-built packages either from [ROCm's package servers](https://rocm.github.io/install.html#installing-from-amd-rocm-repositories) or by clicking the github releases tab and manually downloading, which could be newer.  Release notes are available for each release on the releases tab.
+Download pre-built packages either from [ROCm's package servers](https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html) or by clicking the github releases tab and manually downloading, which could be newer.  Release notes are available for each release on the releases tab.
```
[the remark this answers is not in the document]
Comment (tfalders) [reply README.md:5]: This part of the readme I copied from hipBLAS. It looks like you're right - I didn't see any packages in the hipBLAS releases. I'll update it for hipSOLVER and probably open a PR for hipBLAS as well.

**Comment by cgmb** [comment]
Just FYI, the static analysis build was temporarily disabled globally, which is why the Jenkins docs are not available.

**Review by jzuniga-amd — APPROVED** [review]
LGTM

**Pull request merged**